### PR TITLE
Series of changes from wire.com

### LINKED ISSUE #61: Merging our fork to upstream
Hello,

Our company forked your repo to work with certval and since then we've made many commits on the [fork](https://github.com/carl-wallace/rust-pki/compare/main...wireapp:rust-pki:wire/stable). We would like to merge our changes to upstream, use the main repo and archive ours.

What would be the best approach to do so? We have 25 commits with some fixes, tests and some features that we needed.

### PR DESCRIPTION
**Pull request by augustocdias**
Closes #61

Here's a rebased version of our changes. There were conflicts in almost all conflicts. I hope I resolved them correctly.

### CODE REVIEW

File: certval/src/util/pdv_utilities.rs [hunk 1]
```diff
@@ -7,10 +7,7 @@ use core::str::FromStr;
 
 use log::{debug, error};
 
-#[cfg(feature = "std")]
 use lazy_static::lazy_static;
-
-#[cfg(feature = "std")]
```
Comment (carl-wallace) [inline certval/src/util/pdv_utilities.rs:13]: I am not tracking this change and similar in this file. When I build with `cargo hack build --feature-powerset` it fails straightaway when building `cargo build --no-default-features`. I did not look at regex source code, but nothing in the docs indicates no-std support presently. How did you envision removing the feature gate here working?

**Comment by carl-wallace** [comment]
Aside from not seeing how the removal of the std feature gate for regex and url usage should work, this all looks great. Thank you for doing the work and for sharing. I plan to merge as-is tomorrow but you can expect a few changes over the next days/weeks, including:

- Restore the std feature gate around regex and url usage. Any additional info on what you were aiming for here would be helpful.
- Add an rsa feature gate to tests that require rsa, which is most of them, and, longer-term, replace or augment those tests with artifacts that don't use RSA.
- Update documentation to reflect changes to features, algorithms, etc.
- Update ml-dsa and slh-dsa support to FIPS standard versions once pqcrypto is updated (and/or move to RustCrypto implementations).
- Align with changes to RustCrypto/formats. These changes will likely impact your code (though probably mostly mechanical changes). Many of the changes emanate from https://github.com/RustCrypto/formats/issues/1486.

Once updates for the crates related to the last item are released, I'll likely merge this branch to main and prepare a release of certval for crates.io.

**Comment by augustocdias** [comment]
Hi. Thank you very much. I don't think I'll be able to answer why that was done. Our colleagues that made those changes are not in the company anymore. I'm currently very busy with other tasks and unable to investigate. I'd say go for it and if we see something breaks when we upgrade we come back and discuss what happened.

**Comment by istankovic** [comment]
@carl-wallace is there a plan to release the changes done here? We would very much like to drop our fork. :-)

**Comment by carl-wallace** [comment]
Yes. I am just waiting for the various release candidates that are depended upon to get released.

**Comment by istankovic** [comment]
> Yes. I am just waiting for the various release candidates that are depended upon to get released.

Could you provide more details about those dependencies?
We'd be happy to help out also elsewhere in order to unblock a release here.

**Comment by carl-wallace** [comment]
The patches in the top level Cargo.toml file on the cert-limbo branch (or more broadly the c7 branch) give a view of what all is moving as of a few weeks back. cert-limbo will be the target for release, though I'd like to get the PQ stuff out shortly after. There were some new pre-releases cut in the past few days for the RustCrypto/formats repos but it will be a couple of weeks before I can give them a look (personal health issue).

**Comment by baloo** [comment]
I regularly publish branches with the newest pre-releases

https://github.com/baloo/rust-pki/tree/baloo/bump-2025-08-03 should be the last one.

**Comment by istankovic** [comment]
> The patches in the top level Cargo.toml file on the cert-limbo branch (or more broadly the c7 branch) give a view of what all is moving as of a few weeks back. cert-limbo will be the target for release, though I'd like to get the PQ stuff out shortly after. There were some new pre-releases cut in the past few days for the RustCrypto/formats repos but it will be a couple of weeks before I can give them a look (personal health issue).

Hi, I just wanted to ask about the status here -- I see there is the new `pre-release` branch, does that mean that a new release is close? :sweat_smile:

**Comment by carl-wallace** [comment]
I really don't want to do a release against the released RustCrypto formats and have been waiting on x509_cert v0.3.x and friends. Unfortunately, it's been about two years since the v0.3 work started. I do periodically align the cert_limbo branch with the latest (as does @baloo, whose fork tends to be more current). I have no basis for predicting when a release will be possible.

**Comment by baloo** [comment]
Everything is ready to do at that point, we'll wait for a release of rand_core 0.10 before a final release.

**Comment by istankovic** [comment]
rand_core 0.10 is out, are we still waiting for x509_cert 0.3?

**Comment by carl-wallace** [comment]
Yes. I caught everything up a couple of weeks ago so lag once released should be fairly minimal. @baloo bumped a couple of crates to stable releases yesterday but I've not merged yet.

**Pull request merged**